File: jetpack_java/src/main/java/com/kunminx/jetpack_java/sample_04_databinding/ui/state/DetailViewModel.java


package com.kunminx.jetpack_java.sample_04_databinding.ui.state;

import androidx.databinding.ObservableField;
import androidx.lifecycle.ViewModel;

import com.kunminx.jetpack_java.common_data.bean.Moment;

/**
 * Create by KunMinX at 2020/5/30
 */
public class DetailViewModel extends ViewModel {

    public final ObservableField<String> imgUrl = new ObservableField<>();
    public final ObservableField<String> name = new ObservableField<>();
    public final ObservableField<String> content = new ObservableField<>();
    public final ObservableField<String> location = new ObservableField<>();
    public final ObservableField<String> avatar = new ObservableField<>();

    public void initState(Moment moment) {
        avatar.set(moment.getUserAvatar());
        name.set(moment.getUserName());
        content.set(moment.getContent());
        imgUrl.set(moment.getImgUrl());
        location.set(moment.getLocation());
    }
}

File: jetpack_kotlin/src/main/java/com/flywith24/jetpack_kotlin/sample_04_databinding/ui/state/DetailViewModel.kt
package com.flywith24.jetpack_kotlin.sample_04_databinding.ui.state

import androidx.databinding.ObservableField
import androidx.lifecycle.ViewModel
import com.flywith24.jetpack_kotlin.common_data.bean.Moment

/**
 * @author Flywith24
 * @date   2020/5/31
 * time   23:11
 * description
 */
class DetailViewModel : ViewModel() {
    val imgUrl = ObservableField<String>()
    val name = ObservableField<String>()
    val content = ObservableField<String>()
    val location = ObservableField<String>()
    val avatar = ObservableField<String>()

    fun initState(moment: Moment) {
        avatar.set(moment.userAvatar)
        name.set(moment.username)
        content.set(moment.content)
        imgUrl.set(moment.imgUrl)
        location.set(moment.location)
    }
}